### PR TITLE
feat(turbo-ignore) add a --max-buffer option for dry runs that have large outputs

### PR DESCRIPTION
**Pull request by mehulkar**
Closes TURBO-1371

### CODE REVIEW

File: packages/turbo-ignore/src/ignore.ts [hunk 1]
```diff
@@ -95,50 +95,54 @@ export function turboIgnore(
...
+
+  const execOptions: { cwd: string; maxBuffer?: number } = {
+    cwd: root,
+  };
+
+  if (opts.maxBuffer) {
+    execOptions.maxBuffer = opts.maxBuffer;
+  }
+
+  info(`execOptions: ${JSON.stringify(execOptions, null, 2)}`);
```
Comment (chris-olszewski) [inline packages/turbo-ignore/src/ignore.ts:107]: I don't think we want to be logging this to all users

File: packages/turbo-ignore/src/cli.ts [hunk 2]
```diff
@@ -30,6 +30,12 @@ turboIgnoreCli
       "The directory to run in (default: cwd)"
     )
   )
+  .addOption(
+    new Option(
+      "-b, --max-buffer <number>",
+      "maxBuffer for the child process in KB (default: 1024 * 1024)"
+    ).argParser((val) => parseInt(val, 10))
```
Comment (chris-olszewski) [inline packages/turbo-ignore/src/cli.ts:37]: If I'm reading the docs correctly, I'm pretty sure this is in bytes:

>maxBuffer [<number>](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Number_type) Largest amount of data in bytes allowed on stdout or stderr. If exceeded, the child process is terminated and any output is truncated. See caveat at [maxBuffer and Unicode](https://nodejs.org/api/child_process.html#maxbuffer-and-unicode). Default: 1024 * 1024

[source](https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback)

```suggestion
      "maxBuffer for the child process in KB (default: 1024 KB)"
    ).argParser((val) => parseInt(val, 10) * 1024)
```

File: packages/turbo-ignore/src/types.ts [hunk 3]
```diff
@@ -23,4 +23,6 @@ export interface TurboIgnoreOptions {
   task?: string;
   // A ref/head to compare against if no previously deployed SHA is available
   fallback?: string;
+  // The maxBuffer for the child process in KB
```
Comment (chris-olszewski) [inline packages/turbo-ignore/src/types.ts:26]:
```suggestion
  // The maxBuffer for the child process in bytes
```

**Review by chris-olszewski — APPROVED** [review]
I think it makes sense to give users the ability to control the number of bytes in either KB or MB. I think MB is slightly nicer, but I realize for some it won't be granular enough.

File: packages/turbo-ignore/src/types.ts [hunk 4]
```diff
@@ -23,4 +23,6 @@ export interface TurboIgnoreOptions {
   task?: string;
   // A ref/head to compare against if no previously deployed SHA is available
   fallback?: string;
+  // The maxBuffer for the child process in bytes
```
Comment (mehulkar) [inline packages/turbo-ignore/src/types.ts:26]:
```suggestion
  // The maxBuffer for the child process in KB
```

**Pull request merged**